index.go

```diff
@@ -195,22 +196,26 @@ func main() {
 
 	})
 
+	//Cache should definitely be locked down
 	m.Group("/cache", func(r martini.Router) { // different endpoint because partial matching matches this to another excused route
 		r.Get("/key", cache.GetByKey)
 		r.Get("/keys", cache.GetKeys)
-		r.Delete("/keys", cache.DeleteKey)
+		r.Delete("/keys", middleware.InternalKeyAuthentication, cache.DeleteKey)
 	})
 
+	//No lockdown of customer related endpoints for now
 	m.Group("/cust", func(r martini.Router) { // different endpoint because partial matching matches this to another excused route
 		r.Post("/user/changePassword", customer_ctlr.ChangePassword)
```

Comment:
From the logs it looks like almost all instances of this are 500s, and none of them have keys at all. So not sure.